doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -808,34 +808,34 @@ combined to create custom rewriting procedures. Its set of strategies is based
 on the programmable rewriting strategies with generic traversals by Visser et al.
 :cite:`Luttik97specificationof` :cite:`Visser98`, which formed the core of
 the Stratego transformation language :cite:`Visser01`. Rewriting strategies
-are applied using the tactic :n:`rewrite_strat @rewstrategy`.
+are applied using the :tacn:`rewrite_strat` tactic`.
 
-.. insertprodn rewstrategy rewstrategy
+.. insertprodn rewstrategy rewstrategy_atom
 
 .. prodn::
-   rewstrategy ::= @one_term
-   | <- @one_term
-   | fail
-   | id
-   | refl
-   | progress @rewstrategy
-   | try @rewstrategy
-   | @rewstrategy ; @rewstrategy
-   | choice {+ @rewstrategy }
-   | repeat @rewstrategy
-   | any @rewstrategy
-   | subterm @rewstrategy
+   rewstrategy ::= <- @one_term
    | subterms @rewstrategy
+   | subterm @rewstrategy
    | innermost @rewstrategy
    | outermost @rewstrategy
    | bottomup @rewstrategy
    | topdown @rewstrategy
+   | progress @rewstrategy
+   | try @rewstrategy
+   | any @rewstrategy
+   | repeat @rewstrategy
+   | old_hints @ident
    | hints @ident
    | terms {* @one_term }
    | eval @red_expr
    | fold @one_term
-   | ( @rewstrategy )
-   | old_hints @ident
+   | choice {+ @rewstrategy_atom }
+   | @rewstrategy_atom
+   rewstrategy_atom ::= @one_term
+   | id
+   | fail
+   | refl
+   | ( {| @rewstrategy | {+; @rewstrategy } } )
```

Comment:
This is redundant, right?  It should just be
```suggestion
   | ( {+; @rewstrategy } )
```

@jfehrle what do we need to make this happen?  (Or do you already know how to do this @herbelin ?)

Reply:
I don't know how to control how this is produced.

Reply:
@jfehrle what do we need in doc/tools/docgram/common.edit_mlg to fix this (and the one above?)